src/systems/wheel_slip/WheelSlip.cc

```diff
@@ -242,9 +245,30 @@ bool WheelSlipPrivate::Load(const EntityComponentManager &_ecm,
 /////////////////////////////////////////////////
 void WheelSlipPrivate::Update(EntityComponentManager &_ecm)
 {
-  for (const auto &linkSurface : this->mapLinkSurfaceParams)
+  for (auto &linkSurface : this->mapLinkSurfaceParams)
   {
-    const auto &params = linkSurface.second;
+    auto &params = linkSurface.second;
+    const auto * wheelSlipCmdComp =
+      _ecm.Component<components::WheelSlipCmd>(linkSurface.first);
+    if (wheelSlipCmdComp) {
+      const auto & wheelSlipCmdParams = wheelSlipCmdComp->Data();
+      bool changed = math::equal(
+          params.slipComplianceLateral,
+          wheelSlipCmdParams.slip_compliance_lateral(),
+          1e-6) &&
+        math::equal(
+          params.slipComplianceLongitudinal,
+          wheelSlipCmdParams.slip_compliance_longitudinal(),
+          1e-6);
+
+      if (changed) {
```

Comment:
```suggestion
      if (changed)
      {
```

Reply:
yes, I'm not sure how I didn't notice this.
Thanks for the comment!

Reply:
Fixed in https://github.com/ignitionrobotics/ign-gazebo/pull/1241/commits/25a11e63ce5b3e1486f62996b937f6bdf868c56a

Reply:
```suggestion
      if (changed)
      {
```